ci/infra/testrunner/platforms/terraform.py

```diff
@@ -2,143 +2,65 @@
 import json
 import logging
 import os
-import subprocess
 
-from timeout_decorator import timeout
-
-from utils import (Format, step, Utils)
+from platforms.platform import Platform
+from utils import (Format, step)
 
 logger = logging.getLogger('testrunner')
 
 
-class Terraform:
+class Terraform(Platform):
     def __init__(self, conf, platform):
-        self.conf = conf
-        self.utils = Utils(conf)
+        super().__init__(conf)
         self.tfdir = os.path.join(self.conf.terraform.tfdir, platform)
         self.tfjson_path = os.path.join(conf.workspace, "tfout.json")
+        self.tfout_path = os.path.join(self.conf.workspace, "tfout")
         self.state = None
 
-    def _env_setup_cmd(self):
-        """Returns the command for setting up the platform environment"""
-        return ""
-
-    def _cleanup_platform(self):
-        """Platform specific cleanup. Expected to be overridden by platforms"""
+        self.logs["files"] += ["/var/run/cloud-init/status.json",
+                               "/var/log/cloud-init-output.log",
+                               "/var/log/cloud-init.log"]
 
-    def _get_platform_logs(self):
-        """Platform specific logs to collect. Expected to be overridden by platforms"""
-        return False
-
-    def cleanup(self):
-        """ Clean up """
-        try:
-            self._cleanup_platform()
-        except Exception as ex:
-            cleanup_failure = True
-            logger.warning("Received the following error: '{}'\n"
-                           "Attempting to finish cleanup".format(ex))
-            raise Exception("Failure(s) during cleanup") from ex
-        finally:
-            dirs = [os.path.join(self.conf.workspace, "tfout"),
-                self.tfjson_path]
-            for dir in dirs:
-                try: 
-                    self.utils.runshellcommand("rm -rf {}".format(dir))
-                except Exception as ex:
-                    logger.warning("Received the following error: '{}'\n"
-                                   "Attempting to finish cleanup".format(ex))
-
-    @timeout(600)
-    @step
-    def gather_logs(self):
-        logging_errors = False
+        self.tmp_files = [self.tfout_path,
+                          self.tfjson_path]
 
-        node_ips = {"master": self.get_nodes_ipaddrs("master"),
-                    "worker": self.get_nodes_ipaddrs("worker")}
-        logs = {"files": ["/var/run/cloud-init/status.json",
-                          "/var/log/cloud-init-output.log",
-                          "/var/log/cloud-init.log"],
-                "dirs": ["/var/log/pods"],
-                "services": ["kubelet"]}
+    def destroy(self, variables):
+        cmd = "destroy -auto-approve"
 
-        if not os.path.isdir(self.conf.log_dir):
-            os.mkdir(self.conf.log_dir)
-            logger.info(f"Created log dir {self.conf.log_dir}")
+        for var in variables:
+            cmd += f" -var {var}"
```

Comment:
Does `{var}` need any quoting / escaping here?

Reply:
I would assume that anyone using this would do that where it's being used. Not always good to assume though I guess.